usaspending_api/references/v2/views/filter_tree/psc_filter_tree.py

```diff
@@ -3,30 +3,224 @@
+                    "description": object.description,
+                    "count": self.get_count([object.code], object.code),
+                    "children": None,
+                }
+            )
+        return retval
+
+    def tier_1_search(self, ancestor_array, filter_string, lower_tier_nodes=None) -> list:
+        filters = [Q(Q(Q(length=1) & Q(code__in=PSC_GROUPS["Service"]["terms"])) | Q(length=2))]
+        query = Q()
+        if ancestor_array:
+            parent = ancestor_array[0]
+            filters.append(Q(code__iregex=PSC_GROUPS.get(parent, {}).get("pattern") or "(?!)"))
+        if lower_tier_nodes:
+            lower_tier_codes = [node["id"][:-1] for node in lower_tier_nodes]
+            lower_tier_codes = list(dict.fromkeys(lower_tier_codes))
+            for code in lower_tier_codes:
+                query |= Q(code=code)
+        if filter_string:
+            query |= Q(Q(code__icontains=filter_string) | Q(description__icontains=filter_string))
+        filters.append(query)
+        retval = []
+        for object in PSC.objects.filter(*filters):
+            ancestors = []
+            if object.code.isdigit():
+                ancestors.append("Product")
+            elif object.code[0] in PSC_GROUPS["Research and Development"]["terms"]:
+                ancestors.append("Research and Development")
+            else:
+                ancestors.append("Service")
+            retval.append(
+                {
+                    "id": object.code,
+                    "ancestors": ancestors,
+                    "description": object.description,
+                    "count": self.get_count([object.code], object.code),
+                    "children": None,
+                }
+            )
+        return retval
+
+    def toptier_search(self, filter_string, tier1_nodes=None):
+        retval = []
+        if not filter_string and not tier1_nodes:
+            return [
+                {"id": key, "ancestors": [], "description": "", "count": self.get_count([], key), "children": None}
+                for key in PSC_GROUPS.keys()
+            ]
+        if tier1_nodes:
+            toptier_codes = [node["id"][:1] for node in tier1_nodes]
+            for key in PSC_GROUPS.keys():
+                if set(toptier_codes).intersection(set(PSC_GROUPS[key]["terms"])):
+                    retval.append(
+                        {
+                            "id": key,
+                            "ancestors": [],
+                            "description": "",
+                            "count": self.get_count([], key),
+                            "children": None,
+                        }
+                    )
+        return retval
+
+    def _combine_nodes(self, upper_tier, lower_tier):
+        for node in upper_tier:
+            children = []
+            node_ids = [x["id"] for x in node["children"]] if node["children"] is not None else []
+            for node1 in lower_tier:
```

Comment:
```suggestion
            for lower_node in lower_tier:
```
Or something like that. Meaningful variable names are nice.